### PR TITLE
VMware: Tag management to VMware object

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/vmware_rest_client.py
lib/ansible/modules/cloud/vmware/vmware_tag_manager.py

##### ANSIBLE VERSION

```paste below
2.8-devel
```

### CODE REVIEW

**Comment by ghost** [comment]
Hi,
I'm not an expert in python, so I may be wrong when I read your module. It seems to me that it lacks an important feature. In VMWare, two tags can have the same name if there are in a different tag categories.

I've developed a similar plugin for the company where I work, but to avoid this problem we added the following things: 
* an optional argument with the tag category name 
* if this argument is set we search for tags only in this category
* else we search in all tags 
* but if there are more than one tags with the same name, the ansible role is in error with a message to invite to specify the category. 

I'm not sure if it's the right place to say this, but I didn't see any issue about this topic. 

Regards

**Comment by Akasurde** [comment]
@Un-Cornichon Thanks for sharing your thoughts, which make sense to me. Could you point me to location of source of the module that you mentioned so that I can include that functionality in this module ? 
Thanks.

**Review by pdellaert — APPROVED** [review]
LGTM, i guess integration tests are not possible yet because vcsim doesn't support the REST client?

@Un-Cornichon has a good point :)

File: lib/ansible/modules/cloud/vmware/vmware_tag_manager.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  vmware_tag_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    validate_certs: no
+    tag_names:
+      - Sample_Tag_0002
+    object_name: Fedora_VM
+    object_type: VirtualMachine
+    state: add
```
Comment (dgmorales) [inline lib/ansible/modules/cloud/vmware/vmware_tag_manager.py:72]:
```suggestion
    action: add
```

File: lib/ansible/modules/cloud/vmware/vmware_tag_manager.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  vmware_tag_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    validate_certs: no
+    tag_names:
+      - Sample_Tag_0002
+    object_name: Fedora_VM
+    object_type: VirtualMachine
+    state: remove
```
Comment (dgmorales) [inline lib/ansible/modules/cloud/vmware/vmware_tag_manager.py:85]:
```suggestion
    action: remove
```

**Review by dgmorales — COMMENTED** [review]
@Akasurde I've just tested this (it worked!) and noticed the examples are citing "state" instead of "action".

File: lib/ansible/modules/cloud/vmware/vmware_tag_manager.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_tag_manager.py:72]: Thanks.

**Pull request merged**